internal/node/package_manager/restore.go

```diff
@@ -11,17 +9,12 @@ type PackageManagerRestorer struct {
 	packageManager PackageManager
 }
 
-func (r *PackageManagerRestorer) Restore(restoreEvent *hdb.StateUpdate) error {
-	var nodeState node.NodeState
-	err := json.Unmarshal(restoreEvent.NewState, &nodeState)
-	if err != nil {
-		return err
-	}
-
+func (r *PackageManagerRestorer) Restore(restoreEvent hdb.StateUpdate) error {
+	nodeState := restoreEvent.NewState().(*node.NodeState)
```

Comment:
why is this getting cast? shouldn't the return type be `node.NodeState`?

Reply:
For the same reason as above, there is weirdness because I want the state machine system to be reusable for community schemas down the line.